specs/networking/libp2p-standardization.md

```diff
@@ -135,24 +161,369 @@ number of shard subnets is defined via `SHARD_SUBNET_COUNT` and the shard
 Each Gossipsub
 [Message](https://github.com/libp2p/go-libp2p-pubsub/blob/master/pb/rpc.proto#L17-L24)
 has a maximum size of 512KB (estimated from expected largest uncompressed block
-size).
+size). Clients SHOULD reject messages that are over this size limit.
 
 The `data` field of a Gossipsub `Message` is an SSZ-encoded object. For the `beacon_block` topic,
 this is a `beacon_block`. For the `beacon_attestation` topic, this is
 an `attestation`.
 
-## Eth-2 RPC
-
-#### Protocol Id: `/eth/serenity/beacon/rpc/1`
-
-The [RPC Interface](./rpc-interface.md) is specified in this repository.
-
 ## Discovery
 
 Discovery Version 5
 ([discv5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5.md))
-will be used for discovery. This protocol uses a UDP transport and specifies
-its own encryption, ip-discovery and topic advertisement. Therefore, it has no
+is used for discovery. This protocol uses a UDP transport and specifies
+its own encryption, IP-discovery and topic advertisement. Therefore, it has no
 need to establish streams through `multistream-select`, rather, act
 as a standalone implementation that feeds discovered peers/topics (ENR-records) as
-`multiaddrs` into the libp2p service.
+`multiaddrs` into the libp2p service. The libp2p service subsequently forms
+connections and streams with discovered peers.
+
+## Eth-2 RPC
+
+*There is a specification for the [RPC Interface](./rpc-interface.md) in this
+repository which has implementations for libp2p and non-libp2p client. This
+section focuses solely on the libp2p specification and provides significant
+libp2p-specific modifications - as suggested by @Zah.*
+
+
+### Specification
+
+#### Protocol Segregation
+
+Each RPC message is segregated into it's own libp2p protocol id, which is a string of the form:
+
+```
+<ProtocolPrefix>/<MessageName>/<SchemaVersion>/<Encoding>
+```
+
+With:
+* **ProtocolPrefix** -- the RPC messages are grouped into families identified
+  by a shared LibP2P protocol name prefix. A conforming implementation is
+  expected to support either all messages appearing within a family or none of
+  them. In this case, we use `/eth/serenity/rpc`.
+* **MessageName** -- each RPC request is identified by a name consisting of
+English letters, digits and underscores (_).
+* **SchemaVersion** -- a semantic version consisting of one or more numbers
+  separated by dots (.). Each schema is versioned to facilitate backward and
+  forward-compatibility when possible.
+* **Encoding** -- while the schema defines the data types in more abstract
+  terms, the encoding describes a specific representation of bytes that will be
+  transmitted over the wire. See the [Encodings](#encodings) section, for
+  further details.
+
+
+This protocol segregation allows libp2p `multistream-select` to handle the
+RPC-type, version and encoding negotiation before establishing the underlying
+substreams.
+
+
+#### Requests and Responses
```

Comment:
If I get it right, it is suggested that each request-response would need to open a new dedicated stream. 
Wouldn't this cause a significant overhead in terms of latency, traffic and client computational resources?
Opening a stream requires additional negotiation roundtrip when negotiating a protocol and may required more with multiple protocol versions. 
I'm also not sure how 'lightweight' is stream creation in different libp2p implementations.

Reply:
Yes, in terms of latency; little in terms of traffic and computational resources. I pointed this out here: https://github.com/ethereum/eth2.0-specs/pull/1281#issuecomment-512212450.

Reply:
This was also briefly discussed in #1158. As @raulk mentions there will be overheads in multistream select 1 and potential versioning issues/features where each RPC has it's own version/protocol-id.
We opted for a forward-looking design, where we assume that multistream-select2 is adopted and the overhead is minimized. This doesn't mitigate a potential versioning issue, but we gain some benefits, such as not sending the method_id and encodings in each RPC request. This is of course still up for debate.

Reply:
@raulk Will multistream-2 be able to handle nested protocols in one round like `/secio/1.0.0/mplex/2.6.0` ?